### DISCUSSION TITLE
Feature request: Extend expression body to event

### DISCUSSION THREAD

**Discussion by Orace** [dotnet/csharplang#1903] [Language Ideas]
Consider this .Net 3.5 code:

    interface ISubject
    {
        int Value { get; set; }
        event Action<int> ValueChanged;
    }

    class Wrapper : ISubject
    {
        ISubject subject;

        public int Value
        {
            get { return subject.Value; }
            set { subject.Value = value; }
        }

        public event Action<int> ValueChanged
        {
            add { subject.ValueChanged += value; }
            remove { subject.ValueChanged -= value; }
        }
    }

The `get/set` and `add/remove` code are just boilerplates code.

C# 6.0 has introduced expression-bodied members:

    class Wrapper : ISubject
    {
        ISubject subject;

        public int Value
        {
            get => subject.Value;
            set => subject.Value = value;
        }

        public event Action<int> ValueChanged
        {
            add => subject.ValueChanged += value;
            remove => subject.ValueChanged -= value;
        }
    }

In this case not really a big improvement.
The subject of this proposition is to reduce the code used for the event to this:

        public event Action<int> ValueChanged => subject.ValueChanged;

**Comment by Orace** [comment]
Much forward something more generic to delegate simply a property getter and/or setter to another property is missing.

Actually only the readonly property case is simplifiable with:

        public int Getter => subject.Getter;

Something like:

        public int Value <=> subject.Value;
        public int Value { get; set; } is subject.Value;
        ...

Is really missing.

**Comment by yaakov-h** [comment]
`public int Value <=> subject.Value` is a duplicate feature request. See #1007.
